src/develop/develop.c

```diff
@@ -1038,6 +1038,31 @@ static void _dev_add_history_item_ext(
   }
   if((module->enabled) && (!no_image))
     module->write_input_hint = TRUE;
+
+  // This could be the place to check for automatic history writing
+  const double user_delay = (double)dt_conf_get_int("autosave_interval"); 
+  const double now = dt_get_wtime();
+
+  static double last = 0.0;
+  static gboolean proper_timing = TRUE;
+
+  if((user_delay >= 1.0) && proper_timing && ((now - last) > user_delay))
+  {
+    // Ok, lets save status for image
+    dt_dev_write_history(dev);
```

Comment:
I would put all this code into a separate routine. This will be cleaner and will separate all the code for the auto save support. If i'm not mistaken the only param to this routine is `dt_develop_t`.

I would recommend also using a db transaction around `dt_dev_write_history()` as the first operation is to remove the whole history for the given image. So for an auto-save we really want to be safe.

Reply:
Tears in my eyes :-) I don't speak neither french nor SQL so would that be
```
sqlite3_stmt *stmt = NULL;
dt_database_start_transaction(darktable.db);
dt_dev_write_history(dev);
sqlite3_finalize(stmt);
dt_database_release_transaction(darktable.db);
```

Reply:
No, probably just:

```
dt_database_start_transaction(darktable.db);
dt_dev_write_history(dev);
dt_database_release_transaction(darktable.db);
```

We must just check that there is no other transaction done during `dt_dev_write_history(dev)` call.

Reply:
Should that not be done in dt_dev_write_history (or _ext) itself so that other places that call it also benefit?

Reply:
@dterrahe : Probably so, you're right.

Reply:
I think it should be here for exactly the reason **not** being used elsewhere, 😊

Reply:
Will have a look again...

Reply:
@jenshannoschwalm : You can unconditionally call `dt_database_start_transaction(darktable.db)` in `dt_dev_write_history(dev)` as only the top-level call will start a transaction. I did that some time ago but forgot, actually I tried to use save points, but that was not correct/safe.

Reply:
@jenshannoschwalm : Wait I'm wrong... The start of the transaction is unconditional actually.

`if(trxid == 0 || TRUE)
`

Not sure why... Don't remember all the details, let me review this.

Reply:
@jenshannoschwalm : See #po/db-tx

Reply:
You probably want to rebase your PR on top of #14950. If all goes well I'll merge #14950 before.

Reply:
> @TurboGit the code so far is pretty trivial, if you want to take over feel free to do so :-)

Just seeing your comment here. Ok, let me merge my PR and yours and push that if all is working fine together.